src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -51,14 +56,41 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
           return;
 
         }
-        request.setAttribute("user", user);
+        request.setAttribute("userToAccess", user);
         request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request, response);
   }
 
   
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-        
+        String username = (String) request.getSession().getAttribute("user");
+    if (username == null) {
+      // user is not logged in, don't let them edit profile
+      return;
+    }
+
+    User user = userStore.getUser(username);
+    if (user == null) {
+      // user was not found, don't let them edit profile
+      return;
+    }
+
+    // processor needed for BBCode to HTML translation
+    TextProcessor processor = BBProcessorFactory.getInstance().createFromResource("kefirbb.xml");
+
+    String messageContent = request.getParameter("info");
+
+    // this removes any HTML from the message content and then parses any BBCode to HTML
+    String cleanedMessageContent = Jsoup.clean(messageContent, Whitelist.none());
+
+    String cleanedAndEmojiCheckedMsg = EmojiParser.parseToUnicode(cleanedMessageContent);
+
+    user.changeAboutMe(cleanedAndEmojiCheckedMsg);
```

Comment:
You might want to consider storing what the user typed in and doing the conversion at display time. My main thinking here is that by doing the conversion here, the user will end up seeing the converted output when they try to edit it again. That would likely be surprising to users and a source of frustration.

Reply:
The converted message is displayed on the user's profile if you take a look at the profile.jsp & in the edit box, the non-converted message is shown so that users don't get frustrated :)